### DISCUSSION TITLE
Enable vxlan in guest kernel in Testing Environment

### DISCUSSION THREAD

**Discussion by redbrick9** [strongswan/strongswan#1652] [Q&A] [answered]
Hi folks,

I would like to enable vxlan in guest kernel. I'm trying to modify testing/config/kernel/config-6.1 and change line "# CONFIG_VXLAN is not set" to "CONFIG_VXLAN=y", does it work?

Thanks,
redbrick9

**Comment by tobiasbrunner** [comment, marked as answer]
Possibly. The build script runs `make oldconfig`, which, I guess, should fix inconsistencies. But enabling via `make menuconfig` might work better (build the kernel once with the original config, then switch to the build dir and run that command and afterwards `make`).

**Comment by redbrick9** [comment]
Thank you very much, I will give it a try and update the thread.

**Reply by redbrick9** [reply]
@tobiasbrunner We have verified "make menuconfig", it works too. One more question, how testing/config/kernel/config-6.1 is generated? Thanks!

**Reply by tobiasbrunner** [reply]
It's just the `.config` file from the kernel's build dir after it has been built.

**Reply by redbrick9** [reply]
Oh, maybe I asked an unclear question. There are a few lines of the file, so I'm wondering how the file is generated?

```
   1 #
   2 # Automatically generated file; DO NOT EDIT.
   3 # Linux/x86 6.1.1 Kernel Configuration
   4 #
   5 CONFIG_CC_VERSION_TEXT="gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0"
```

**Reply by tobiasbrunner** [reply]
That's added/updated by the kernel's Kconfig system (e.g. if you run `make oldconfig` or `make menuconfig`).

**Reply by redbrick9** [reply]
Understood, thanks again.
